File: CSE/velcheva-viktoriya/front-end/signup.js
let user = window.localStorage.getItem('user');
if (!user) {
    user = {
        id: undefined,
        first_name: undefined,
        last_name: undefined,
        email: undefined,
        password: undefined,
        phone: undefined,
        dob: undefined,
        gender: undefined,
        created_at: undefined,
        updated_at: undefined,
        is_active: undefined
    }
} else {
    user = JSON.parse(user);
}

function signupStep1() {
    const password = document.getElementById('password');
    const confirmPassword = document.getElementById('confirm_password');
    if (password.value == "" || (password.value != confirmPassword.value)) {
        console.log("passwords are different");
        return;
    }

    const terms = document.getElementById('terms');
    if (!terms.checked) {
        console.log("terms policies must be agreed upon");
        return;
    }
    const privacy = document.getElementById('privacy');
    if (!privacy.checked) {
        console.log("privacy policies must be agreed upon");
        return;
    }

    user.first_name = document.getElementById('firstname').value;
    user.last_name = document.getElementById('lastname').value;
    user.email = document.getElementById('email').value;
    user.phone = document.getElementById('phone').value;
    user.password = password.value;
    
    const dob = document.getElementById('dob');
    const gender = document.getElementById('gender');
    user.dob = new Date(dob.value).getTime();
    user.gender = parseInt(gender.value);
    console.log(user);
    
    // fetch(url, options).then().then().catch()
    // Create new user
    const url = "http://localhost:5001/signup"
    const options = {
        method: "POST",
        headers: {
            "Content-Type": "application/json"
        },
        body: JSON.stringify(user)
    };
    fetch(url, options).then(responseArrived).then(responseBodyReceived).catch(errorHappened);

    // window.localStorage.setItem('user', JSON.stringify(user));
    // window.location.replace("signup-2.html");
}

function signupStep2() {
   
}

const form = document.querySelector('form');
form.addEventListener('submit', stopFormDefault);

function stopFormDefault(event) {
    event.preventDefault();
    // actual logic, e.g. validate the form
    console.log('Form submission cancelled.');
}

function getData() {
    // fetch(url, options).then(f(x)).then(g(x)).catch(h(x))
    const url = "http://localhost:5001/version"
    const options = {
        method: "GET",
        headers: {
            "Content-Type": "application/json"
        }
    }

    fetch(url)
    .then(responseArrived)
    .then(responseBodyReceived)
    .catch(errorHappened);
    console.log("Hello!!");
}

function responseArrived(response) {
    if (!response.ok) {
        throw new Error("Failed to get data.");
    }
    return response.json();
}

function responseBodyReceived(response) {
    if (response.data && response.data.id) {
        window.localStorage.setItem('user-id', response.data.id);
        window.localStorage.setItem('user-first-name', response.data.first_name);
        window.location.replace("signup-2.html");
    } else {
        console.log("Error: User data is missing from the response");
    }
}

function errorHappened(response) {
    console.log("Error!!!");
    console.log(response);
}

File: CSE/velcheva-viktoriya/project/backend/signup.py
from datetime import datetime
from flask import Flask, request, redirect, jsonify
from storage import connect  # Импорт на функцията за свързване с базата данни
from entities import User  # Импорт на User класа

app = Flask(__name__)

@app.route('/signup', methods=['POST'])
def signup():
    data = request.json  # Взимаме JSON данните от заявката

    # Създаваме нов User обект с въведените данни
    new_user = User()
    new_user.first_name = data.get("first_name")
    new_user.last_name = data.get("last_name")
    new_user.email = data.get("email")
    new_user.phone = data.get("phone")
    new_user.password = data.get("password")
    new_user.dob = data.get("dob")
    new_user.gender = data.get("gender")
    new_user.created_at = datetime.now().isoformat()
    new_user.updated_at = datetime.now().isoformat()
    new_user.is_active = 1

    # Проверка на валидността на паролата
    if not new_user.is_password_valid():
        return jsonify({"error": "Password must be at least 8 characters"}), 400

    # Вмъкване на новия потребител в базата данни
    try:
        connection = connect()
        new_user.insert(connection)
        return jsonify({"message": "User registered successfully"}), 200
    except Exception as e:
        print(f"Error: {e}")
        return jsonify({"error": "Failed to register user"}), 500

if __name__ == "__main__":
    app.run(debug=True)
